ff-asn-apply-filters/index.html

```diff
@@ -0,0 +1,284 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>ASN "Apply Filters" Demo</title>
+  <link rel="stylesheet" href="../assets/demo-styles.css">
+
+  <script src="../node_modules/ff-web-components/dist/vendor/custom-elements-es5-adapter.js"></script>
+  <script src="../node_modules/ff-web-components/dist/vendor/webcomponents-loader.js"></script>
+  <script defer src="../node_modules/ff-web-components/dist/bundle.js"></script>
+
+  <style>
+    main {
+      display: flex;
+      padding: 0 10px;
+    }
+
+    .filters {
+      flex: 0 0;
+      width: 240px;
+      margin-right: 10px;
+    }
+
+    .products-listing {
+      flex: 1 0;
+    }
+
+    ff-record-list {
+      margin-right: -10px;
+    }
+  </style>
+
+  <script>
+    const difference = (a, b) => a.filter(x => !b.includes(x));
+    const symmetricDifference = (a, b) => difference(a, b).concat(difference(b, a));
+
+    function createButton() {
+      const button = document.createElement('button');
+      button.textContent = 'Apply filters';
+      button.classList.add('btn');
+      button.enable = () => {
+        button.disabled = false;
+        button.classList.add('btn--pending');
+      }
+      button.disable = () => {
+        button.disabled = true;
+        button.classList.remove('btn--pending');
+      }
+      return button;
+    }
+
+    const currentSelection = eventAggregator => filter => {
+      const values = eventAggregator.currentSearchResult.filters.find(f => f.name === filter) || {valueList: []};
+      return values.valueList.map(v => v.value);
+    }
+
+    const pendingSelection = grp => {
+      return Array.from(grp.querySelectorAll('ff-asn-group-element') || []).filter(g => g.selected).map(g => g.element);
```

Comment:
[OPTIONAL]
`|| []` is not strictly necessary because `querySelectorAll` always returns an array.
```suggestion
      return Array.from(grp.querySelectorAll('ff-asn-group-element')).filter(g => g.selected).map(g => g.element);
```